src/pages/api/secure/preferences/payment/reset-pin.ts

```diff
@@ -0,0 +1,89 @@
+import jwt from 'jsonwebtoken'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { withSessionRoute } from '@/ironAuth/withSessionApiRoute'
+import { VerificationChannel } from '@/types/AccountNotifications'
+import {
+  updatePaymentPreferences,
+  verifyVerificationCode,
+} from '@/utils/database'
+
+interface ResetPinBody {
+  newPin: string
+  token: string
+}
+
+interface TokenPayload {
+  type: string
+  account_address: string
+  iat: number
+  exp: number
+  jti: string
+}
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'POST') {
+    return res.status(405).json({ error: 'Method not allowed' })
+  }
+
+  const JWT_SECRET = process.env.JWT_SECRET
+  if (!JWT_SECRET) {
+    return res
+      .status(500)
+      .json({ error: 'JWT_SECRET environment variable is required' })
```

Comment:
Not useful to users, We should not mention JWT to users. This should be "There was an issue scheduling updating preferences. Please get in touch with us through support@meetwith.xyz"